x/genutil/module.go

```diff
@@ -0,0 +1,101 @@
+package genutil
+
+import (
+	"encoding/json"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+var (
+	_ sdk.AppModule      = AppModule{}
+	_ sdk.AppModuleBasic = AppModuleBasic{}
+)
+
+const moduleName = "genutil"
+
+// app module basics object
+type AppModuleBasic struct{}
+
+// module name
+func (AppModuleBasic) Name() string {
+	return moduleName
+}
+
+// module name
+func (AppModuleBasic) RegisterCodec(cdc *codec.Codec) {}
+
+// module name
+func (AppModuleBasic) DefaultGenesis() json.RawMessage { return nil }
+
+// module validate genesis
+func (AppModuleBasic) ValidateGenesis(bz json.RawMessage) error {
+	var data GenesisState
+	err := moduleCdc.UnmarshalJSON(bz, &data)
+	if err != nil {
+		return err
+	}
+	return ValidateGenesis(data)
+}
+
+//___________________________
+// app module
+type AppModule struct {
+	AppModuleBasic
+	accountKeeper AccountKeeper
+	stakingKeeper StakingKeeper
+	cdc           *codec.Codec
+	deliverTx     deliverTxfn
+}
+
+// NewAppModule creates a new AppModule object
+func NewAppModule(accountKeeper AccountKeeper, stakingKeeper StakingKeeper,
```

Comment:
So we _require_ the staking keeper here? Isn't this `genutil` module supposed to be a bit more general?

Reply:
k so just to be clear because we are using expected keepers, genutils require \*a\* staking keeper not _the_ staking keeper. This is necessary for applying validator set updates after the genesis-txs have been processed, which I think makes sense short of moving all the genesis-transaction logic into staking which I don't think makes sense. So this is fine AFAICS

Reply:
Agree. Just wanted to be clear on what was going on here. Thank you for the clarification!